radiotools/coreas/generate_coreas_sim.py

```diff
@@ -241,19 +415,118 @@ def write_list_star_pattern(filename, zen, az, append=False, obs_level=1564.0, o
+    
+    print("Saved antenna positions (in groundplane coordinates) to file: ", filename)
+
     fout.close()
 
+    # in case you want to plot the antennas in the shower plane coordinate system
+    # mainly for visual checking whether the starshape is ok
+    if vxB_plot:
+        # open the shower.list file to save the generated starshapes to
+        with open("shower_plane.list", "w") as file:
+                
+            # transform the station positions to vxB system for plot
+            shower_plane_system = cs.transform_to_vxB_vxvxB(np.array(station_positions_groundsystem))
+         
+            for i in range(len(shower_plane_system)):
+                # save the generated starshapes to the antenna.list file
+                # positions in cm
+                file.write(f"AntennaPosition = {shower_plane_system[i, 0]} {shower_plane_system[i, 1]} {shower_plane_system[i, 2]} {name}\n")
+            
+            print("Saved antenna positions (in vxB_vxvxB coordinates) to file: ", "shower_plane.list")
+
+
+    # return corsika azimuth angle to for automatically generating corsika input files with the right values
+    return corsika_azimuth
+
+
+def get_rmax(X_obs):
+    """ 
+    WARNING: this is just a very rough approximation in order to generate antenna patterns
+    of the right size for starshape simulations
+
+
+    Parameters:
+
+    X_obs: Grammage at observation level (in g/cm^2)
+
+    Returns: 
+        Returns maximum axis distance in meter for a given simulation as
+        function of the grammage at observation level X_obs in g/cm2 for a given atmosphere and zenith angle
+    """
+    # rough hardcoded parametrisation...
+    return -148 + 0.712 * X_obs
+
+
+def get_starshaped_pattern_radii(zenith, obs_level, n0=1.000292, at=None, atm_model=None):
+    """
+    function to generate starshape antenna pattern with certain features:
+
+    Dense core 
+
+    Dense rings up to just outside the cherenkov ring
+
+    Sparse up to multiple times of the cherenkov radius
+    """
+
+    # This is just validated for has shower
+    # is not even sopisticated
+
+    # obs_level has to be given in m
+    # zenith must be given in radians
+
+    # errors that catch when input is in wrong unit
+    if obs_level > 10000:
+            sys.exit(f"Observation level likely given in cm: {obs_level}. Must be given in meters!")
+    
+    if zenith > 7:
+            sys.exit("Zenith angle likely given in degrees. Must be given in radians!")
+
+    if at is None:
+        if atm_model is None:
+            sys.exit("No proper arguments for get_starshaped_pattern_radii")
+
+        at = models.Atmosphere(atm_model)
+
+    # calculate maximum distance of antenna from shower core (in shower plane)
+    # uses rough, hardcoded parametrisation
+    maxX = at.get_atmosphere(zenith, obs_level)
+    rmax = get_rmax(maxX)
```

Comment:
Where is this coming form? Is that a (very) old parameterisation from Christian based on vertical showers? I would tend to define a maximal radius by a multiplicity of cherenkov radii.

Reply:
I think so. It's effectively doing that by generating antenna rings that stop about 5 Cherenkov radii out. I just did that with a little trial and error. I agree though that it's more elegant to use the Cherenkov radius directly, especially with the new functions I added. 
I'll make the changes soon.